src/main/java/com/amazon/opendistroforelasticsearch/security/auditlog/routing/AsyncStoragePool.java

```diff
@@ -72,11 +59,18 @@ public void submit(AuditMessage message, AuditLogSink sink) {
 		}
 	}
 
-	private ThreadPoolExecutor createExecutor(final int threadPoolSize, final int maxQueueLen) {
+	private ThreadPoolExecutor createExecutor(final AsyncStoragePoolConfig config) {
```

Comment:
Not sure why this method is necessary, can't it be inlined in the constructor?

Reply:
This class will be subscribed to a config listener in future PR which will update for any config changes which will call this function again. Hence moved it to be passed in a method.

Reply:
To support config changes, it would be necessary to properly handle existing pool shutdown and creation of a new thread pool. What if there are pending tasks to execute? What if a running task takes too long to complete? This method is likely to require significant refactoring and new functionality before it can support config change.

Reply:
Thread pool size can be dynamically changed. However dynamically changing the queue length will be more challenging as the current queue must depleted as LinkedBlockingQueue size cant be changed once. For sure. it will need more work. 
This PR just cleans up exisiting code.

Reply:
Please clarify, if you want to create another Executor on configuration change, it would be necessary to deal with the issues I mentioned. If configuration change will modify an existing Executor, this function can't be reused as it creates new instance of `ThreadPoolExecutor`. I'd recommend to inline the function, especially that it looks like it was supposed to be `static` (it takes member variable `config` as an argument), but also requires access to `log` that is not static.

Reply:
I was answering your concern of handling existing queue when there is a config change. I passed this as a param because createExecutor looked like a candidate that can be made separate. I will make this static.